pkg/controller/sshd/sshd_controller.go

```diff
@@ -630,41 +585,21 @@ func newSSHDService(cr *cloudingressv1alpha1.SSHD) *corev1.Service {
 	}
 }
 
-func (r *ReconcileSSHD) ensureDNSRecords() error {
-	// Private zone
-	for i := 1; i <= config.MaxAPIRetries; i++ {
-		err := r.awsClient.UpsertARecord(
-			r.route53.privateHostedZoneName,
-			r.route53.loadBalancerDNSName,
-			r.route53.loadBalancerHostedZoneId,
-			r.route53.resourceRecordSetName,
-			"RH SSH Endpoint", false)
-		if err != nil {
-			log.Info("Couldn't upsert a DNS record for private zone: " + err.Error())
-			if i == config.MaxAPIRetries {
-				log.Info("Out of retries for private zone")
-				return err
-			}
-			log.Info(fmt.Sprintf("Sleeping %d seconds before retrying...", i))
-			time.Sleep(time.Duration(i) * time.Second)
-		} else {
-			break
-		}
+func (r *ReconcileSSHD) ensureDNSRecords(cr *cloudingressv1alpha1.SSHD) error {
+	// fetch the sshd service
+	svc := &corev1.Service{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: cr.Namespace, Name: cr.Name}, svc)
+	if err != nil {
+		return err
 	}
 
-	// Public zone
+	// ensure records
 	for i := 1; i <= config.MaxAPIRetries; i++ {
-		// Append a dot to get the zone name.
-		err := r.awsClient.UpsertARecord(
-			r.route53.publicHostedZoneName,
-			r.route53.loadBalancerDNSName,
-			r.route53.loadBalancerHostedZoneId,
-			r.route53.resourceRecordSetName,
-			"RH SSH Endpoint", false)
+		err := r.cloudClient.EnsureSSHDNS(context.TODO(), r.client, cr, svc)
 		if err != nil {
-			log.Info("Couldn't upsert a DNS record for public zone: " + err.Error())
+			log.Info("Couldn't ensure DNS record for zone: " + err.Error())
 			if i == config.MaxAPIRetries {
-				log.Info("Out of retries for public zone")
+				log.Info("Out of retries for private zone")
 				return err
```

Comment:
I had assumed the retry loop for "ensure" and "delete" DNS methods had been moved into `CloudClient` but it appears not.  @lisa: Should they be?  Strikes me as implementation detail controllers shouldn't have to deal with.